### PR TITLE
Implement globals

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: lib/fizzy/execute.cpp [hunk 1]
```diff
@@ -154,11 +154,18 @@ Instance instantiate(const Module& module)
...
     // NOTE: fill it with zeroes
     bytes memory(pages * pagesize, 0);
-    return {module, std::move(memory)};
+    // TODO: add imported globals first
+    std::vector<uint64_t> globals;
+    globals.reserve(module.globalsec.size());
+    for (auto const& global : module.globalsec)
+        globals.emplace_back(global.init_value);
+
+    return {module, std::move(memory), globals};
```
Comment (axic) [inline lib/fizzy/execute.cpp]: Move globals?

File: lib/fizzy/execute.cpp [hunk 2]
```diff
@@ -154,11 +154,25 @@ Instance instantiate(const Module& module)
...
+    // TODO: add imported globals first
+    std::vector<uint64_t> globals;
+    globals.reserve(module.globalsec.size());
+    for (auto const& global : module.globalsec)
+    {
+        if (global.init_type == GlobalInitType ::constant)
+            globals.emplace_back(global.init.value);
+        else
+        {
+            // TODO: initialize by imported global
```
Comment (axic) [inline lib/fizzy/execute.cpp]: Can you throw a runtime exception here?

File: lib/fizzy/execute.cpp [hunk 3]
```diff
@@ -172,11 +172,27 @@ Instance instantiate(const Module& module)
     }
     // NOTE: fill it with zeroes
     bytes memory(memory_min * page_size, 0);
-    return {module, std::move(memory), memory_max};
+
+    // TODO: add imported globals first
+    std::vector<uint64_t> globals;
+    globals.reserve(module.globalsec.size());
```
Comment (axic) [inline lib/fizzy/execute.cpp]: Can't you just use `std::vector<uint64_t> globals(module.globalsec.size())` ?

File: test/unittests/execute_test.cpp [hunk 4]
```diff
@@ -117,6 +117,42 @@ TEST(execute, local_tee)
     EXPECT_EQ(ret[0], 42);
 }
 
+TEST(execute, global_get)
+{
+    fizzy::Module module;
+    fizzy::Global g{true, fizzy::GlobalInitType::constant, {42}};
+    module.globalsec.emplace_back(g);
```
Comment (axic) [inline test/unittests/execute_test.cpp]: Can you squash these two?

File: lib/fizzy/execute.cpp (hunk 3 above)
Comment (gumb0) [reply lib/fizzy/execute.cpp]: there's no constructor that reserves but not resizes

File: test/unittests/execute_test.cpp (hunk 4 above)
Comment (gumb0) [reply test/unittests/execute_test.cpp]: It won't compile, because it can't deduce the type of `{42}`

Comment (axic) [reply test/unittests/execute_test.cpp]:
```
module.globalsec.emplace_back(fizzy::Global{true, fizzy::GlobalInitType::constant, {42}});
``

won't compile ?

Comment (gumb0) [reply test/unittests/execute_test.cpp]: this one will, thanks

File: lib/fizzy/parser.cpp [hunk 5]
```diff
@@ -21,6 +21,68 @@ struct parser<FuncType>
...
 };
 
+template <>
+struct parser<Global>
+{
+    parser_result<Global> operator()(const uint8_t* pos)
+    {
+        ValType type;
+        std::tie(type, pos) = parser<ValType>{}(pos);
+        if (type != ValType::i32 && type != ValType::i64)
```
Comment (chfast) [inline lib/fizzy/parser.cpp]: parser<ValType> will throw this error

Comment (gumb0) [reply lib/fizzy/parser.cpp]: removed

**Pull request merged**